src/test/apps.test.js

```diff
@@ -1090,3 +1091,69 @@ describe("App Status component", function () {
   });
 
 });
+
+describe("Breadcrumb Component", function () {
+  before(function () {
+    this.renderComponent = (group, app, task) => {
+      var renderer = TestUtils.createRenderer();
+      renderer.render(
+        <BreadcrumbComponent group={group} app={app} task={task} />
+      );
+      return renderer.getRenderOutput();
+    };
+  });
+
+  it("shows root path by default", function () {
+    var component = this.renderComponent();
+    var rootLink = component.props.children[0].props.children;
+    expect(rootLink.props.to).to.equal("apps");
+    expect(rootLink.props.children).to.equal("Applications");
+  });
+
+  it("renders group names correctly", function () {
+    var component = this.renderComponent("/group-a/group-b/group-c/");
+    var groupItems = component.props.children[1];
+    var linkText = groupItems
+      .filter((li) => !!li)
+      .map((li) => li.props.children.props.children);
```

Comment:
It's a good point. Not much nicer but slightly more readable I guess:
`.map((li) => ShallowUtils.getText(li.props.children));`